pkg/daemon/ovs.go

```diff
@@ -471,14 +473,26 @@ func configureNic(link, ip string, macAddr net.HardwareAddr, mtu int) error {
 	return nil
 }
 
-func configExternalBridge(provider, bridge string) error {
+func configExternalBridge(provider, bridge, nic string) error {
 	output, err := ovs.Exec(ovs.MayExist, "add-br", bridge)
 	if err != nil {
-		return fmt.Errorf("failed to create bridge %s, %v: %q", bridge, err, output)
+		return fmt.Errorf("failed to create OVS bridge %s, %v: %q", bridge, err, output)
 	}
-	output, err = ovs.Exec(ovs.IfExists, "get", "open", ".", "external-ids:ovn-bridge-mappings")
-	if err != nil {
-		return fmt.Errorf("failed to get external-ids, %v", err)
+	if output, err = ovs.Exec("list-ports", bridge); err != nil {
+		return fmt.Errorf("failed to list ports of OVS birdge %s, %v: %q", bridge, err, output)
+	}
+	if output != "" {
+		for _, port := range strings.Split(output, "\n") {
+			if port != nic {
```

Comment:
why remove other ports here

Reply:
1. If an OVS bridge needs multiple ports, maybe the ports should be bonded into a bond interface or use OVS bonding;
2. Currently add events and update events of provider networks are handled in the same function(s), so other ports are considered as old bridged ones and should be removed from the bridge.

Reply:
If I remember correctly, ovn will create patch port on provider bridge

Reply:
You are right. I ignored patch ports. What do you think about removing all other ports excluding the patch ones?